### ISSUE TITLE
[Question] How to properly include modules in your script?

### ISSUE THREAD

**Issue by Bluscream** [pathmann/pyTSon#14]
Lets say i found a nice python module on google that sounds promising to make my script less cluttered etc.
1. Do i just `import` it? Or do i have to download it and place it in some folder?
2. If so, what do i need? Just the source or do i need to ship binaries for every platform i want to deploy on?
3. If the needed files contain a dll, can i ship it in a .ts3_plugin package or does it need to be manually moved like pytsons python35.dll ?

P.S. Why is the sqlite3 module working fine for me but for noone else even though they downloaded the sqlite3 dll and placed it in their ts folder?

**Comment by pathmann** [comment]
1. If the module is in the python standard library: yes, import it. Otherwise the includes subfolder is the right place for it. It's in sys.path by default. But of course other directories might work as well if you adapt sys.path in your script.
2. That depends on the type of python module. If it's pure python, see above. If your module consists of a c plugin (or c plugin + python code), you have to deliver that. I'm not quite sure, if this works out of the box from the includes directory. Some parts of the standard library are of type 2 (eg. ctypes) and it works here (see Lib and Lib/lib-dynload subfolders). Feel free to test and report back, if anything fails, I'll have a look.
3. Some parts of this question (and answer) are windows only: No, the python library is linked to the plugin, so this dependency is required on loading. It's the ts3 client's job to load plugins and it does not care about directories. So linked dependencies have to be in the client's root directory. Python modules are loaded by the embedded python interpreter, which in this case does care about directories (which are currently includes, includes/Lib, includes/Lib/lib-dynload and scripts; all subdirectories of ~/plugins/pyTSon/).
So in short: the answer is "yes", you can deliver python dependencies in a ts3 plugin package.
